### PR TITLE
Add support for arm64 builds for charm-local-users

### PR DESCRIPTION
**Pull request by samuelallan72**
This ports the release.yaml changes from charm-local-users to here. :)

### CODE REVIEW

**Comment by samuelallan72** [comment]
From my refactor https://github.com/canonical/solutions-engineering-automation/pull/38 :

> type any will be required if we want template variables of other types - for example a list of architectures to run on (useful for multiarch releases coming soon)

Unfortunately it turned out this didn't actually work, so we need to use string values for everything with the current implementation anyway.

**Review by jneo8 — APPROVED** [review]

File: terraform-plans/templates/github/charm_check.yaml.tftpl [hunk 1]
```diff
@@ -64,6 +64,7 @@ jobs:
         with:
           provider: "lxd"
           juju-channel: $${{ matrix.juju-channel }}
+          charmcraft-channel: "2.x/stable"
```
Comment (chanchiwai-ray) [inline terraform-plans/templates/github/charm_check.yaml.tftpl:67]: is it really called `2.x/stable`?

**Review by chanchiwai-ray — APPROVED** [review]
A question, otherwise LGTM

File: terraform-plans/templates/github/charm_check.yaml.tftpl (hunk 1 above)
Comment (samuelallan72) [reply terraform-plans/templates/github/charm_check.yaml.tftpl:67]: Yes

```
$ snap info charmcraft  | grep 2.x/stable
  2.x/stable:       2.7.1                 2024-07-23 (4162) 52MB classic
```

**Pull request merged**